roboto-flex: init at 3.200

romildo opened the pull request:
The [roboto](https://github.com/google/roboto) font repository has not been updated since 2017, and recently it has been archived.

[Roboto Flex](https://github.com/googlefonts/roboto-flex) is its continuation, in a brand new repository.

See https://github.com/googlefonts/roboto-flex for more information.

Maintainers of packages that depend on roboto should consider migrating them to using roboto-flex.

Closes https://github.com/NixOS/nixpkgs/issues/353596

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Atemu reviewed (approved):
Diff LGTM.

When ofBorg is done, please ping me again.

Atemu commented:
A package apparently depends on roboto, please fix that. See the CI eval error.

romildo commented:
> A package apparently depends on roboto, please fix that. See the CI eval error.

I am not understanding the issue reported by the CI eval error.

Is it related to the use of `rec` in the derivation?

Also after https://github.com/NixOS/nixpkgs/pull/354531 was merged today, I had to redo the undo and redo the changes, in order to solve merge conflicts. May it be related?

Atemu commented:
Another package depends on roboto. You need to adjust all dependent packages' inputs for the new game.

It has nothing to do with `rec`.

Your rebase looks fine to me.

romildo commented:
As there are other packages in nixpkgs that uses `roboto`, and the font family name has changed from `Roboto` to `Roboto Flex`, maybe it is better to not remove the `roboto` package for now, and just add `roboto-flex`.

Then let the maintainers of the packages depending on `roboto` to find out how to migrate their packages to `roboto-flex`.

Atemu commented:
That sounds good to me.

romildo committed:
roboto-flex: init at 3.200

The roboto font repository has not been update since 2017, and
recently it has been archived.

Roboto Flex is its continuation, in a brand new repository.

See https://github.com/googlefonts/roboto-flex for more information.

Maintainers of packages that depend on roboto should consider
migrating them to using roboto-flex.

Atemu commented:
@ofBorg build roboto-flex
